app_dart/lib/src/request_handling/swarming_authentication.dart

```diff
@@ -111,9 +100,57 @@ class SwarmingAuthenticationProvider extends AuthenticationProvider {
 
       return AuthenticatedContext(agent: agent, clientContext: clientContext);
     } else if (swarmingToken != null) {
-      return await authenticateIdToken(swarmingToken, clientContext: clientContext, log: log);
+      return await authenticateAccessToken(swarmingToken, clientContext: clientContext, log: log);
     }
 
     throw const Unauthenticated('Request rejected due to not from LUCI or Cocoon agent');
   }
+
+  /// Authenticate [accessToken] against Google OAuth 2 API.
+  ///
+  /// Access tokens are the legacy authentication strategy for Google OAuth, where ID tokens
+  /// are the new technique to use. LUCI auth only generates access tokens, and must be
+  /// validated against a different endpoint. We only validate access tokens if they belong
+  /// to a DeviceLab service account.
```

Comment:
Should this be updated to also cover the prod luci service account?

Reply:
Done